### DISCUSSION TITLE
how to search/filter on the value of a custom field choice set?

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#17393] [Q&A]
I've read PR's and a lot about how the ability to search inside custom fields for their values has been added to NetBox incrementally over the past few years, but I'm at a loss on how to structure a query to search for the value of a custom field choice set?

If I have a custom field that is a "selection" or "multiple selection" type, how do I filter a search to show me all objects that have the value of something inside a custom field choice set?

For example, if I have a custom field "multiple selection" called "business_units". I have a custom field choice set also called "business units" with the following extra choices:
choice1:bu1
choice2:bu2
choice3:bu3

if I attempt to search on "bu2" in global search I get nothing. I can't see any guidance on how I would create a filter that would allow me to display all objects where "bu2" has been a selected choice.

**Comment by candlerb** [comment]
Such low-cardinality values aren't available in global search, but they're available as filters.

Example: I have some multi-select custom fields on Device. If I browse to devices and then click the 'filter' tab, I have filters available for these, at the bottom of the page.

If I select one, when I go back to the 'results' tab it takes me to `/dcim/devices/?cf_foo=bar`

**Reply by ghost** [reply]
the goal is to try and pair this with https://github.com/netbox-community/netbox/issues/17394

That way we can sort of replicate what tags do, while having a bit more control over the context of the metadata.
IE, we can make a custom field called xyz, then do selections or multi-selections to give the choices meaning.

rather than a tag called "bu1" .. we can make a custom field called 'business unit', and one of the options is 'bu1'. that way, if https://github.com/netbox-community/netbox/issues/17394 is implemented, we could click on 'bu1' and be taken to a list of all other object that 'bu1' is associated with.
